### PR TITLE
[FEATURE] Human vs. Random Computer

### PR DESCRIPTION
**Pull request by crkubiak**

### CODE REVIEW

File: src/main/java/GameSelector.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                Board boardHVRc = new Board();
+                Validation validationHVRc = new Validation();
+                Rules rulesHVRc = new Rules();
+                RandomComputer randomComputerHVRc = new RandomComputer();
+                Game gameHVRc = new HumVsRandCPUGame(boardHVRc, validationHVRc, rulesHVRc, randomComputerHVRc);
+                gameHVRc.individualGameLoop();
+                break;
+        }
+    }
+}
```
Comment (scottyplunkett) [inline src/main/java/GameSelector.java]: Is there any difference between the board, validation, and rule objects in these options?
If not you should try moving their instantiation to the beginning of the method to D.R.Y. up the code, or remove them entirely by using an overloaded constructor in the Game implementations.
example:
```
public class GameSelector {
    public void gameSelection(int gameType) {
        Game configuration;
        Board board = new Board();
        Validation validation = new Validation();
        Rules rules = new Rules();
        if (gameType == 1) {
                configuration = new HumVsHumGame(board, validation, rules);
        } else {
                RandomComputer computer = new RandomComputer();
                configuration = new HumVsRandCPUGame(board, validation, rules, computer);
        }
        configuration.individualGameLoop();
    }
}
```

Comment (scottyplunkett) [reply src/main/java/GameSelector.java]: You should also consider what this class has been made responsible for.
Should a class that sets up the appropriate game configuration also be responsible for starting the game loop?

**Review by scottyplunkett — CHANGES_REQUESTED** [review]

**Comment by scottyplunkett** [comment]
What dependencies need to change between different types of games?
Maybe, the only Game dependency that changes is the type of players. And furthermore maybe the only thing that changes between different types of players is the strategy or logic they use to figure out what move they are going to make.
Try to refactor this so that: 
1. `Player` becomes an object dependent on a Strategy. ex. _Human, Easy_ 
2. `GameSelector` becomes `Config` and the only thing it is responsible for is setting up the player objects based on options selected by a user. (i.e. _Config injects Strategy dependency into Players -> TicTacToe injects Config dependency into Game_)

Nice work getting this working, look forward to seeing how you refactor. Let me know if you have any questions.

File: src/main/java/GameSelector.java (hunk 1 above)
Comment (crkubiak) [reply src/main/java/GameSelector.java]: I was wondering if I could do that, I think I said it on the phone call yesterday but wasn't saying it clearly.  I've moved the repetitive dependencies to the top of the class and made them private.

I changed the variable to Game configuration, I then returned configuration at the end of the switch statement, and the game loop is called in the TicTacToe.java runner.

Comment (crkubiak) [reply src/main/java/GameSelector.java]: 1. Completed refactor switch statement
2. Injecting player classes into the game directly as a dependency
3. Cleaned up testing and notes
4. Removed human movement logic from Game and into Human

**Review by scottyplunkett — APPROVED** [review]

**Pull request merged**